src/array.ts

```diff
@@ -3,6 +3,14 @@ import { wrapNative } from './support/util';
 import { forOf, isArrayLike, isIterable, Iterable } from './iterator';
 import { MAX_SAFE_INTEGER as maxSafeInteger } from './number';
 
+/**
+ * An ArrayLike object
+ */
+export interface ArrayLike<T> {
```

Comment:
So this actually causes issues for downstream modules consumers, `typings install` in `core` using symlinks errors for - 

`Attempted to compile "dojo-shim" as a global module, but it looks like an external module. You'll need to remove the global option to continue.`

We cannot  remove the `export` as `ArrayLike<T>` is a private name, so we can use a different name for this local interface or just cast occurrences where global `ArrayLike`'s typings are too aggressive, example" `target[offset] = target[start];` --> `(<any> target)[offset] = target[start];`